tools/roslaunch/src/roslaunch/loader.py

```diff
@@ -499,6 +499,17 @@ def param_value(self, verbose, name, ptype, value, textfile, binfile, command):
                     # special handle the use of Python scripts in Windows environment:
                     # 1. search for a wrapper executable (of the same name) under the same directory with stat.S_IXUSR flag
                     # 2. if no wrapper is present, prepend command with 'python' executable
+                    def split_with_single_quote_enclosed_fixup(command):
+                        in_command = shlex.split(command, posix=False)  # use non-posix method on Windows
+                        if not ("'" in command):
+                            return in_command
+                        new_command = []
+                        for token in in_command:
+                            if token.startswith("'") and token.endswith("'"):
+                                new_command.append("{}".format(token[1:-1]))
+                            else:
+                                new_command.append(token)
+                        return new_command
```

Comment:
this does not seem right

normal code path returns a string if on Windows, this is changing `command` from `str` to `str[]`

Reply:
Yes, it does return `str[]` now but it is an acceptable data type for [`Popen`](https://docs.python.org/2/library/subprocess.html#popen-constructor). Any special case I should be aware of?

Reply:
I don't remember the exact reason why we were using string instead of array of array

a tricky case would be to verify when the input `command` already contains `"` in it. Since we're passing a list of string, we'd be relying on `subprocess.list2cmdline` (`Popen` uses it to re-construct the list of string into a single string) and `shlex.split(, posix=False)`. When `command` already contains `"`, `shlex.split(, posix=False)` seems to keep those in the result:

```python
>>> shlex.split('"path to executable" --flag arg', posix=False)
['"path to executable"', '--flag', 'arg']
```

and `subprocess.list2cmdline` escapes the double quotes again, making the final command to be

```python
'"\\"path to executable\\"" --flag arg'
```

and this could lead to errors. I am not sure when this hit, it might be because roslaunch uses Windows' find to resolve its parameters

related discussion: https://stackoverflow.com/a/39096422